### ISSUE TITLE
Missing line segment after slicing non-manifold mesh

### ISSUE THREAD

**Issue by sweco-sekrsv** [julienr/meshcut#20]
Hi!

Thanks for a great library.

In `examples/1_stl_sphere_cut.py`, with the settings

```
example_fname = os.path.join('data', 'temp.stl')
plane_orig = (387.224, 232.544, 6.287)
plane_norm = (0, 1, 0)
```

I have attached the` temp.stl` file, see below.

I get two lines from the slicing but I expect a single continuous line. One piece of the line is missing.
I had a look at the triangle where the missing line is and I can see nothing wrong with it. See this screenshot:

![meshcut_error](https://user-images.githubusercontent.com/10772529/145015634-dacdc94e-573e-4322-8759-2cebb3962ef5.png)

It seems I might have the same or similar problem as in this issue:
https://github.com/julienr/meshcut/issues/7

[temp.zip](https://github.com/julienr/meshcut/files/7667695/temp.zip)

**Comment by julienr** [comment]
Thanks for reporting and providing example data.

After looking at the issue, it may indeed be related to #7 and more generaly, one assumption in meshcut is that all meshes are watertight. This is an important assumption because right now, meshcut works as follow:
1. find all intersections between plane and mesh triangles
2. generate polylines by starting at one random intersection and 'walk' the polyline

Now, for non-watertight mesh (like you have here), the "start at a random intersection" part of 2 doesn't work anymore because if we start walking the polyline from an intersection that's in the middle of the mesh, we'll get two disjoint polylines. I think that's what is happening here.

I don't have a solution just yet, but Ill think about it a bit more. At least the problem is better identified.

**Comment by julienr** [comment]
This should have been closed by #21 . I've added the mesh you provided as a regression test for this issue.

Please test the fix and let me know if it doesn't work on some other cases.

(I haven't yet updated the pypi package, so you'll have to install from git master)
